src/core/thread/mle_router.cpp

```diff
@@ -2035,6 +2035,7 @@ ThreadError MleRouter::HandleChildUpdateRequest(const Message &aMessage, const I
     }
 
     child->mLastHeard = Timer::GetNow();
+    child->mAddSrcMatchEntryShort = true;
```

Comment:
Hi @jwhui , the reason for not choosing Data Request with a short source address as an indicator is that Child Update Request will use extended address as source and dest, and when queueing Child Update Response message, the Data Request with a short source address may still not received(which has 100ms delay). Thus extended address will be added into the source match table, and the following Data Request with a short source address will be Acked with FramePending unset, the radio will go into sleep, and the parent will remove the child neighbor because the Child Update Response could not be Acked.

Here when `HandleChildUpdateRequest` from child, it could be sure that the child already has a short address, and following Data Request must be using short address. So `mAddSrcMatchEntryShort` is set to be true here.

Any thoughts?